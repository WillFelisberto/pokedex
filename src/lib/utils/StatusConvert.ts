// Converte altura de decÃ­metros para metros
export const convertDecimetresToMeters = (heightInDecimetres: number): number => {
  return heightInDecimetres / 10;
};

// Converte peso de hectogramas para quilogramas
export const convertHectogramsToKilograms = (weightInHectograms: number): number => {
  return weightInHectograms / 10;
};

export const formatPokemonId = (id: number): string => {
  return id.toString().padStart(3, '0');
};
