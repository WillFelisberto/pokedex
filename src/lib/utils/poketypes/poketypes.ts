import { pokemonIcons } from '../pokeIconsPng';

export type PokemonType =
  | 'normal'
  | 'fire'
  | 'water'
  | 'electric'
  | 'grass'
  | 'ice'
  | 'fighting'
  | 'poison'
  | 'ground'
  | 'flying'
  | 'psychic'
  | 'bug'
  | 'rock'
  | 'ghost'
  | 'dragon'
  | 'dark'
  | 'steel'
  | 'fairy';

type TypeStyle = {
  iconType: 'outlined' | 'filled';
  iconUrl: string;
  color: string;
};

const getPokemonTypeStyle = (type: PokemonType, isFilled: boolean = false): TypeStyle => {
  const iconType = isFilled ? 'filled' : 'outlined';
  const iconUrl = pokemonIcons[iconType][type] || pokemonIcons.outlined.normal;

  const typeStyles: Record<PokemonType, TypeStyle> = {
    normal: { iconUrl, color: '#919AA2', iconType },
    fire: { iconUrl, color: '#FF9D55', iconType },
    water: { iconUrl, color: '#5090D6', iconType },
    electric: { iconUrl, color: '#F4D23C', iconType },
    grass: { iconUrl, color: '#63BC5A', iconType },
    ice: { iconUrl, color: '#73CEC0', iconType },
    fighting: { iconUrl, color: '#CE416B', iconType },
    poison: { iconUrl, color: '#B567CE', iconType },
    ground: { iconUrl, color: '#D97845', iconType },
    flying: { iconUrl, color: '#89AAE3', iconType },
    psychic: { iconUrl, color: '#FA7179', iconType },
    bug: { iconUrl, color: '#91C12F', iconType },
    rock: { iconUrl, color: '#C5B78C', iconType },
    ghost: { iconUrl, color: '#5269AD', iconType },
    dragon: { iconUrl, color: '#0B6DC3', iconType },
    dark: { iconUrl, color: '#5A5465', iconType },
    steel: { iconUrl, color: '#5A8EA2', iconType },
    fairy: { iconUrl, color: '#EC8FE6', iconType }
  };

  return typeStyles[type] || { iconUrl: pokemonIcons.outlined.normal, color: '#000000', iconType };
};

export default getPokemonTypeStyle;
