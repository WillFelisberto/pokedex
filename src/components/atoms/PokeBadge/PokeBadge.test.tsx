import { render, screen } from '@/tests/test-utils';
import '@testing-library/jest-dom';

import { PokemonType } from '@/lib/utils/poketypes';

import { PokeBadge } from '.';

describe('<PokeBadge />', () => {
  const type: PokemonType = 'fire';

  it('should render the badge', () => {
    const { container } = render(<PokeBadge type={type} />);
    expect(screen.getByText(/Fire/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the icon', () => {
    render(<PokeBadge type={type} />);
    const icon = screen.getByRole('img', { name: /fire/i });
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveAttribute('src', expect.stringContaining('fire.png'));
  });

  it('should render the text', () => {
    render(<PokeBadge type={type} />);
    expect(screen.getByText('Fire')).toBeInTheDocument();
  });

  it('should have the correct background color', () => {
    render(<PokeBadge type={type} />);
    const badge = screen.getByTestId('poke-badge');
    expect(badge).toHaveClass('bg-pokemon-fire');
  });
});
