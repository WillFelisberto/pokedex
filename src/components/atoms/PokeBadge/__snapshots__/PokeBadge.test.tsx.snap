// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokeBadge /> should render the badge 1`] = `
<div>
  <div
    class="flex w-fit items-center gap-2 text-white px-3 py-2 rounded-full bg-pokemon-fire"
    data-testid="poke-badge"
  >
    <div
      class="w-6 h-6 flex items-center bg-white justify-center rounded-full overflow-visible"
    >
      <img
        alt="fire"
        height="13"
        src="/icons/png/filled/fire.png"
        width="13"
      />
    </div>
    <span
      class="font-semibold text-sm"
    >
      Fire
    </span>
  </div>
</div>
`;
